### ISSUE TITLE
Unable to set profile, getting "not found" 

### ISSUE THREAD

**Issue by ramon-roca** [cflurin/node-red-contrib-ramp-thermostat#26]
Trying a per-room calibration based on a template, while trying to set a profile by sending a JSON formatted string, the thermostat nodes raises an error at the debug window, providing the JSON string followed by a "not found" message.
I can't see what's wrong... the JSON profile that I'm sending looks correct to me.
Below the error, which includes the JSON sent.

```
23/12/2018, 23:51:01 node: Thermostat Despatx
msg : string[140]
"{"name":"Despatx","points":[{"00:00":17.5},{"06:00":17.5},{"11:55":18.5},{"12:01":17},{"22:00":17},{"22:01":18.5},{"23:59":18.5}]} not found"
```

and below, the code that sends de JSON formatted string to the ramp-thermostat node:
```

var progWeekly = flow.get('progWeekly'),
    h = 0;


msg.payload.name = "Despatx";
for (p = 0; p < msg.payload.points.length; p++) 
    for (var h in msg.payload.points[p]) 
        msg.payload.points[p][h] += progWeekly.calibration;

msg.topic = "setProfile";
msg.payload = JSON.stringify(msg.payload);
node.status({text: "Calibration: "+progWeekly.calibration});

return msg;
```

Any ideas / help on what's wrong?

**Comment by ramon-roca** [comment]
TRying to investigate by looking into the code for setProfile at ramp-thermostat.js, looks to me that is expecting an object for setting a profile with an array of points.
But sending the object instead of the JSON string, then I'm getting the error:

```
4/12/2018, 10:17:21node: Thermostat Despatx
msg : error
"SyntaxError: Unexpected token N in JSON at position 5"

```
Although trying, still I'm not seeing whats wrong with the profile object I'm sending, looks correct to me, , I'm not used debugging javascript, so any clue will be very much appreciated!

**Comment by cflurin** [comment]
Hi,

1. which version are you using? You should use the latest v0.8.0
2. yes it is an object
3. see: [Profile-with-more-than-10-points](https://github.com/cflurin/node-red-contrib-ramp-thermostat/wiki/Profile-with-more-than-10-pointsl)

4. Please post your complete flow including  `flow.set ('progWeekly', ...)`

**Comment by ramon-roca** [comment]
Found a workaround: Instead of giving the points as an array of objects [ {"00:00": 19}, ... ], and since at the code I saw that for "backwards compatibility" the code was ready for parsing the points not as an array but as a single object with all the points { "00:00": 19, "06:00": 20,  ... }, I gave the profile in that format, and now works!

Just afraid of using the part of the code that says "backwards compatibility"... and not knowing which should be the current expected format.

**Comment by ramon-roca** [comment]
posted the previous before seeing your comment, sorry.
I'm using 0.8.0, just installed a few days ago.
progWeekly has customized context variables, if you want to see the profile format that I was using, was at the first post of this thread.

**Comment by ramon-roca** [comment]
sorry for the last post, I just pasted the complete flow, which is a bit large, and not sure that I copied properly creating that large post instead of a single line.

more info: trying the example you pointed at: 
https://github.com/cflurin/node-red-contrib-ramp-thermostat/wiki/Profile-with-more-than-10-points

when pressing the "setProfile" injection node, reproduces the same error I was getting in my flow:
```
24/12/2018, 13:03:34node: my room
msg : error
"SyntaxError: Unexpected token N in JSON at position 5"
```

**Comment by cflurin** [comment]
Strange! I've just imported the example and did `setProfile` without any issues.
Could you post the log when starting node-red (os, node-red, node.js versions ...)

Example:
```
Welcome to Node-RED
===================
19 Dec 13:55:52 - [info] Node-RED version: v0.19.5
19 Dec 13:55:52 - [info] Node.js  version: v8.14.0
19 Dec 13:55:52 - [info] Linux 4.14.79-v7+ arm LE
19 Dec 13:55:54 - [info] Loading palette nodes
19 Dec 13:56:02 - [info] Dashboard version 2.12.2 started at /ui
...
````

Note: please add a line with ``` at the begin and at the end when posting logs/code/flows.

**Comment by cflurin** [comment]
... I can't import your flow, so you could delete it (less scroll)

**Comment by ramon-roca** [comment]
deleted
Here it is: the log starting:
```

ec 24 13:37:32 emonpi systemd[1]: Started Node-RED graphical event wiring tool.
Dec 24 13:37:37 emonpi Node-RED[6073]: 24 Dec 13:37:37 - [info]
Dec 24 13:37:37 emonpi Node-RED[6073]: Welcome to Node-RED
Dec 24 13:37:37 emonpi Node-RED[6073]: ===================
Dec 24 13:37:37 emonpi Node-RED[6073]: 24 Dec 13:37:37 - [info] Node-RED version: v0.19.4
Dec 24 13:37:37 emonpi Node-RED[6073]: 24 Dec 13:37:37 - [info] Node.js  version: v8.11.1
Dec 24 13:37:37 emonpi Node-RED[6073]: 24 Dec 13:37:37 - [info] Linux 4.14.71-v7+ arm LE
Dec 24 13:37:40 emonpi Node-RED[6073]: 24 Dec 13:37:40 - [info] Loading palette nodes
Dec 24 13:37:44 emonpi Node-RED[6073]: 24 Dec 13:37:44 - [info] Dashboard version 2.11.0 started at /ui
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [warn] ------------------------------------------------------
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [warn] [node-red-node-pi-sense-hat/sensehat] Error: Can't find Sense HAT python libraries. Run sudo apt-get install sense-hat
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [warn] ------------------------------------------------------
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [info] Settings file  : /home/pi/.node-red/settings.js
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [info] Context store  : 'default' [module=memory]
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [info] User directory : /home/pi/.node-red
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [warn] Projects disabled : editorTheme.projects.enabled=false
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [info] Flows file     : /home/pi/.node-red/flows_emonpi.json
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [info] Server now running at http://127.0.0.1:1880/
Dec 24 13:37:45 emonpi Node-RED[6073]: 24 Dec 13:37:45 - [warn]
Dec 24 13:37:45 emonpi Node-RED[6073]: ---------------------------------------------------------------------
Dec 24 13:37:45 emonpi Node-RED[6073]: Your flow credentials file is encrypted using a system-generated key.
Dec 24 13:37:45 emonpi Node-RED[6073]: If the system-generated key is lost for any reason, your credentials
Dec 24 13:37:45 emonpi Node-RED[6073]: file will not be recoverable, you will have to delete it and re-enter
Dec 24 13:37:45 emonpi Node-RED[6073]: your credentials.
Dec 24 13:37:45 emonpi Node-RED[6073]: You should set your own key using the 'credentialSecret' option in
Dec 24 13:37:45 emonpi Node-RED[6073]: your settings file. Node-RED will then re-encrypt your credentials
Dec 24 13:37:45 emonpi Node-RED[6073]: file using your chosen key the next time you deploy a change.
Dec 24 13:37:45 emonpi Node-RED[6073]: ---------------------------------------------------------------------
Dec 24 13:37:46 emonpi Node-RED[6073]: 24 Dec 13:37:46 - [info] Starting flows
Dec 24 13:37:54 emonpi Node-RED[6073]: 24 Dec 13:37:54 - [info] Started flows
Dec 24 13:37:54 emonpi Node-RED[6073]: 24 Dec 13:37:54 - [info] [mqtt-broker:localhost] Connected to broker: mqtt://127.0.0.1:1883
 
```

**Comment by cflurin** [comment]
It seems to be ok (apart the warn concerning sense-hat).

Try this example without map and chart:

```
[{"id":"4bc2df57.20d66","type":"ramp-thermostat","z":"6da90f72.dff58","name":"my room","profile":"7a5597f.52e6468","hysteresisplus":"1","hysteresisminus":"1","x":360,"y":3900,"wires":[[],[],["89451e27.d569"]]},{"id":"76b5a739.8d6c28","type":"inject","z":"6da90f72.dff58","name":"","topic":"getProfile","payload":"actual","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":160,"y":3920,"wires":[["4bc2df57.20d66"]]},{"id":"ec5fa341.229c9","type":"inject","z":"6da90f72.dff58","name":"setProfile","topic":"setProfile","payload":"{\"name\":\"dining room\",\"points\":[{\"00:00\":18},{\"03:00\":18},{\"06:00\":18.5},{\"08:00\":20},{\"10:00\":20},{\"11:00\":20.5},{\"12:30\":20.5},{\"12:30\":19.5},{\"15:00\":19.5},{\"17:00\":20.5},{\"19:00\":20.5},{\"19:00\":20},{\"21:30\":20},{\"21:30\":18},{\"23:59\":18}]}","payloadType":"json","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":180,"y":3880,"wires":[["4bc2df57.20d66"]]},{"id":"89451e27.d569","type":"debug","z":"6da90f72.dff58","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","x":570,"y":3920,"wires":[]},{"id":"7a5597f.52e6468","type":"profile","z":"","name":"test","time1":"00:00","temp1":"16.1","time2":"07:00","temp2":"17","time3":"19:00","temp3":"20.5","time4":"20:00","temp4":"17","time5":"23:59","temp5":"17","time6":"","temp6":"","time7":"","temp7":"","time8":"","temp8":"","time9":"","temp9":"","time10":"","temp10":""}]
```

1. setProfile
2. getProfile: actual

You should see an object in the debug

**Comment by ramon-roca** [comment]
tested. looks like my java doesn't like it (same JSON parse error):

```
24/12/2018, 16:18:43node: my room
msg : error
"SyntaxError: Unexpected token N in JSON at position 5"

```
To me seems that at at line #377 of ramp-thermostat.js I have nothing at Object.values(point)

**Comment by cflurin** [comment]
I‘ll have a look tomorrow.

**Comment by cflurin** [comment]
Ok, I've replaced `Object.values()`

```
- points[i] = JSON.parse('{"m":' + minutes + ',"t":' + Object.values(point) + '}');
+ points[i] = JSON.parse('{"m":' + minutes + ',"t":' + point[k] + '}');
```
Please, download the latest version from GitHub.

**Comment by ramon-roca** [comment]
works, THANKS!

**Comment by cflurin** [comment]
Just published v0.8.1
